### PR TITLE
fix dangling filters to Swarm

### PR DESCRIPTION
**Pull request by cheyang**
It's very useful to list and remove non used docker images. We  always run

```
docker images -q -a -f dangling=true
```

but when using docker swarm, we notice that there is no logic to handle dangling filter for docker images. And it also indicates that the behavior of docker api and docker swarm api for listing images are different.

[docker images api](https://docs.docker.com/engine/reference/api/docker_remote_api_v1.24/#/list-images)

This is PR to fix the issue and keep the same behavior with docker engine. 
[docker/daemon/images.go](https://github.com/docker/docker/blob/master/daemon/images.go#L41)

### CODE REVIEW

**Comment by xianlubird** [comment]
@dongluochen @nishanttotla @allencloud PTLA, Thanks.

**Comment by nishanttotla** [comment]
@cheyang could you please rebase and force push to restart tests?

**Comment by cheyang** [comment]
@nishanttotla , seems that this error does nothing with my PR. Could you help to take a look, thx.

not ok 63 zk discovery: watch for changes
# (from function `retry' in file discovery/../helpers.bash, line 80,
#  in test file discovery/zk.bats, line 59)
#   `retry 10 1 discovery_check_swarm_info' failed

**Comment by cheyang** [comment]
@nishanttotla Please take a look at  `https://jenkins.dockerproject.org/job/Swarm-PRs%20(engine%201.12)/225/console`
```
not ok 64 zk discovery: node removal
# (from function `retry' in file discovery/../helpers.bash, line 80,
```

Looks like it does nothing with PR.

**Comment by nishanttotla** [comment]
@cheyang thanks, the test failure is unrelated. We'll review this PR.

**Comment by cheyang** [comment]
any update? @nishanttotla

File: cluster/image.go [hunk 1]
```diff
@@ -126,9 +127,33 @@ func (images Images) Filter(opts ImageFilterOptions) Images {
 		return false
 	}
 
+	danglingFilter := func(image *Image) bool {
+		if opts.Filters.Include("dangling") {
+			if len(image.RepoTags) == 0 {
+				image.RepoTags = []string{"<none>:<none>"}
```
Comment (dongluochen) [inline cluster/image.go]: The `Filter` function is to find images that matches filters. It should be read-only, i.e., not changing another data in images.

**Comment by allencloud** [comment]
Thanks for your contribution. @cheyang 
I am wondering what would happen if I execute `docker images -q -a -f dangling=nonexisting`?

**Comment by cheyang** [comment]
Please see PR #2670 .

File: cluster/image.go (hunk 1 above)
Comment (cheyang) [inline cluster/image.go]: Thanks, I've update according to your suggestion. see PR #2670

**Comment by cheyang** [comment]
@allencloud ,

> I am wondering what would happen if I execute docker images -q -a -f dangling=nonexisting?

For `dangling=nonexisting`, this filter will not affect.  

This is the behavior of docker daemon.
···
docker images -q -a -f dangling=nonexisting
Error response from daemon: Invalid filter 'dangling=[nonexisting]'
···

**Pull request closed without merging**